packages/core/base/src/adapter.ts

```diff
@@ -90,14 +90,31 @@ export abstract class BaseWalletAdapter extends EventEmitter<WalletAdapterEvents
 }
 
 export function scopePollingDetectionStrategy(performDetection: () => boolean): void {
+    // Early return when server-side rendering
     if (typeof window === 'undefined' || typeof document === 'undefined') return;
+
+    const onDocumentLoaded = () => {
+        if (performDetection()) {
+            // Wallet detected, nothing more to do
+            return;
+        }
+        // Wallet not detected yet, try detecting every second
+        let repeatingTimer = setInterval(() => {
+            if (performDetection()) {
+                // Wallet detected, nothing more to do
+                clearInterval(repeatingTimer);
+            }
+        }, 1000);
```

Comment:
If the wallet isn't detected, we're in one of two states, right?

1. It hasn't been introduced into the `window` scope yet.
2. It's not installed in the browser, at all.

Unless I'm reading this incorrectly, this will keep polling forever, if we hit case 2, right?

Reply:
Reading back over the description in your PR, I can see that this was your intent :)

Reply:
In general, we should try to avoid scheduling work forever, so the browser process can sleep. There are a couple of things we could add to this to make it lighter on the system:

1. Build a task queue like @jordansexton suggested: https://developer.mozilla.org/en-US/docs/Web/API/Background_Tasks_API
2. Use `requestIdleCallback` if available to schedule work _only_ when the browser is idle.

```
// https://jsbin.com/dadehafeqe/1/edit?js,console
function detect() {
  console.log('do detection work');
}
function schedule() {
  setTimeout(() => {
    requestIdleCallback(() => {
      if (detect()) {
        return;
      }
      schedule();
    });
  }, 1000);
}
if (!detect()) {
  schedule();   
}
```

Reply:
Yea, I agree this definitely isn't the best way to handle this whole situation, but it is _a_ way! I'd much rather see wallets/extensions handle it like we will in Glow, i.e.:
- post a message to `window` once the extension is ready
- catch it in the adapter [like so](https://github.com/solana-labs/wallet-adapter/pull/325/commits/b89d518f11ae7406999888a0208edb531caef032)

This way there's no need to do anything periodically!

Maybe we should:
- add an argument to `scopePollingDetectionStrategy` which would disable the forever-repeating interval
- encourage wallet adapters to implement event-based live detection and disable the interval?